agg-ast/schema_derivation/src/lib.rs

```diff
@@ -198,6 +200,72 @@ pub(crate) fn get_or_create_schema_for_path_mut(
     schema
 }
 
+// this helper simply checks for document schemas and inserts a field with a given schema into
+// that document, ensuring it is required. It follows the same structure as get_or_create_schema_for_path_mut,
+// except that we ignore Schema::Any's.
+fn insert_required_key_into_document_helper(
```

Comment:
After trying to retrofit the existing `get_or_create_schema_for_path_mut` function, I found creating a separate helper to be cleaner in the end. This will create the whole path as required, and only work with documents (it ignores Schema::Any). This prevents the nested documents being created from being `additional_properties: true`, which `get_or_create_schema_for_path_mut` does because it inserts new keys with `Schema::Any`

If a use case arises in the future, we can make the behavior about required optional in the future.